src/daft-functions/src/list/bool_or.rs

```diff
@@ -0,0 +1,55 @@
+use common_error::{DaftError, DaftResult};
+use daft_core::{
+    prelude::{Field, Schema},
+    series::Series,
+};
+use daft_dsl::{
+    functions::{ScalarFunction, ScalarUDF},
+    ExprRef,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct ListBoolOr {}
+
+#[typetag::serde]
+impl ScalarUDF for ListBoolOr {
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn name(&self) -> &'static str {
+        "list_bool_or"
+    }
+
+    fn to_field(&self, inputs: &[ExprRef], schema: &Schema) -> DaftResult<Field> {
+        match inputs {
+            [input] => {
+                let inner_field = input.to_field(schema)?.to_exploded_field()?;
```

Comment:
Similar question as to why we need to call `.to_exploded_field()?` as the one I had for `ListBoolAnd`.

Reply:
Resolved offline. This is used to assert that the field has a datatype of either `List` or `FixedSizeList`.